3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr_config_generator.rb

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) 2012-2017 Snowplow Analytics Ltd. All rights reserved.
+#
+# This program is licensed to you under the Apache License Version 2.0,
+# and you may not use this file except in compliance with the Apache License Version 2.0.
+# You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the Apache License Version 2.0 is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+
+# Author::    Ben Fradet (mailto:support@snowplowanalytics.com)
+# Copyright:: Copyright (c) 2012-2017 Snowplow Analytics Ltd
+# License::   Apache License Version 2.0
+
+require 'contracts'
+require 'iglu-client'
+
+# Implementation of Generator for emr clusters
+module Snowplow
+  module EmrEtlRunner
+    class EmrConfigGenerator
+
+      include Snowplow::EmrEtlRunner::Generator
+      include Snowplow::EmrEtlRunner::Utils
+      include Contracts
+
+      STANDARD_HOSTED_ASSETS = "s3://snowplow-hosted-assets"
+
+      Contract String => Iglu::SchemaKey
+      def get_schema_key(version)
+        Iglu::SchemaKey.parse_key("iglu:com.snowplowanalytics.dataflowrunner/ClusterConfig/avro/#{version}")
+      end
+
+      Contract ConfigHash, Bool, Maybe[String], String, ArrayOf[String] => Hash
+      def create_datum(config, debug=false, resume_from=nil, resolver='', enrichments=[])
+        legacy = (not (config[:aws][:emr][:ami_version] =~ /^[1-3].*/).nil?)
+        region = config[:aws][:emr][:region]
+
+        {
+          "name" => config[:aws][:emr][:jobflow][:job_name],
+          "logUri" => config[:aws][:s3][:buckets][:log],
+          "region" => region,
+          "credentials" => {
+            "accessKeyId" => config[:aws][:access_key_id],
```

Comment:
this is handled by df-runner: https://github.com/snowplow/dataflow-runner/blob/master/src/utils.go#L25-L41